src/twisted/internet/task.py

```diff
@@ -818,12 +811,14 @@ def advance(self, amount):
         time.
         """
         self.rightNow += amount
-        self._sortCalls()
-        while self.calls and self.calls[0].getTime() <= self.seconds():
-            call = self.calls.pop(0)
-            call.called = 1
-            call.func(*call.args, **call.kw)
-            self._sortCalls()
+        while self.calls:
+            time, call = self.calls[0]  # smallest item
```

Comment:
...you'll have to replace this...
```suggestion
            call = heapq.heappop(self.calls)
```